### PR TITLE
gh-120831: Increase the default minimum supported iOS version to 13.0

### LINKED ISSUE #120831: ios buildbot failure: `enclose 'sqlite3_create_window_function' in a __builtin_available check to silence this warning`
This failed in my PR: https://github.com/python/cpython/pull/120442
But, it does not look related.

Link: https://buildbot.python.org/all/#builders/1380/builds/655

```
HEAD is now at 8334a1b55c gh-120384: Fix array-out-of-bounds crash in `list_ass_subscript` (#120442)
Switched to and reset branch 'main'

configure: WARNING: no system libmpdecimal found; falling back to bundled libmpdecimal (deprecated and scheduled for removal in Python 3.15)
configure: WARNING: pkg-config is missing. Some dependencies may not be detected correctly.

configure: WARNING: no system libmpdecimal found; falling back to bundled libmpdecimal (deprecated and scheduled for removal in Python 3.15)
configure: WARNING: pkg-config is missing. Some dependencies may not be detected correctly.

../../Modules/_sqlite/connection.c:1307:14: warning: 'sqlite3_create_window_function' is only available on iOS 13.0 or newer [-Wunguarded-availability-new]
        rc = sqlite3_create_window_function(self->db, name, num_params, flags,
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.5.sdk/usr/include/sqlite3.h:5533:16: note: 'sqlite3_create_window_function' has been marked as being introduced in iOS 13.0 here, but the deployment target is iOS 12.0.0
SQLITE_API int sqlite3_create_window_function(
               ^
../../Modules/_sqlite/connection.c:1307:14: note: enclose 'sqlite3_create_window_function' in a __builtin_available check to silence this warning
        rc = sqlite3_create_window_function(self->db, name, num_params, flags,
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../Modules/_sqlite/connection.c:1315:14: warning: 'sqlite3_create_window_function' is only available on iOS 13.0 or newer [-Wunguarded-availability-new]
        rc = sqlite3_create_window_function(self->db, name, num_params, flags,
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator17.5.sdk/usr/include/sqlite3.h:5533:16: note: 'sqlite3_create_window_function' has been marked as being introduced in iOS 13.0 here, but the deployment target is iOS 12.0.0
SQLITE_API int sqlite3_create_window_function(
               ^
../../Modules/_sqlite/connection.c:1315:14: note: enclose 'sqlite3_create_window_function' in a __builtin_available check to silence this warning
        rc = sqlite3_create_window_function(self->db, name, num_params, flags,
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2 warnings generated.

--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:iOS Simulator, id:C6AEA11C-C34A-47B8-BD67-AF0403ECA353, OS:17.5, name:iPhone SE (3rd generation) }
{ platform:iOS Simulator, id:C6AEA11C-C34A-47B8-BD67-AF0403ECA353, OS:17.5, name:iPhone SE (3rd generation) }
2024-06-21 11:16:45.090 xcodebuild[33218:87319398] [MT] IDETestOperationsObserverDebug: 1075.762 elapsed -- Testing started completed.
2024-06-21 11:16:45.090 xcodebuild[33218:87319398] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
2024-06-21 11:16:45.090 xcodebuild[33218:87319398] [MT] IDETestOperationsObserverDebug: 1075.762 sec, +1075.762 sec -- end
Failing tests:
	-[iOSTestbedTests testPython]

** TEST FAILED **

make: *** [testios] Error 1
```

### Linked PRs
* gh-121250
* gh-121833
* gh-122339
* gh-122341

### PR DESCRIPTION
**Pull request by freakboy3742**
Bumps the default minimum iOS version from 12.0 to 13.0.

As discussed on #120831, the previous default value of 12.0 worked successfully at runtime, but raised a compilation warning about the usage of an SQLite API that wasn't added until iOS 13.0. Although protection exists in the code to prevent any runtime problem, avoiding the compilation warning was judged to be more important.

The usage of iOS 12 is estimated to be less that 0.5% of the total iOS install base, so the impact on the user base is minor. It's also only a *default* version; it's possible to manually configure any minimum version at compile time.

In addition to increasing the default minimum version, this PR modifies the logging generated by the configure script to be more explicit about the iOS deployment target (aka minimum version) that will be used. It also adds usage documentation about the default iOS supported version.

The buildbot will still generate the warning because it the iOS buildbot configuration currently hard-codes the 12.0 minimum supported version. I'll push an update for the buildbot configuration shortly.

Fixes #120831.

* Issue: gh-120831

----
📚 Documentation preview 📚: https://cpython-previews--121250.org.readthedocs.build/

### CODE REVIEW

**Comment by freakboy3742** [comment]
!buildbot iOS

**Comment by freakboy3742** [comment]
!buildbot iOS

File: Doc/using/ios.rst [hunk 1]
```diff
@@ -33,6 +33,17 @@ running, so you only need to deal with the Python code itself.
...
 ========================
 
+iOS version compatibility
+-------------------------
+
+The minimum supported iOS version is specified at compile time, using the
+:option:`--host` option to ``configure``. By default, when compiled for iOS,
+Python will be compiled with a minimum supported iOS version of 13.0. To use a
+different miniumum iOS version, provide the version number as part of the
+:option:`--host` argument - for example,
```
Comment (erlend-aasland) [inline Doc/using/ios.rst]: We try to only link once to the same reference in a paragraph, in order to reduce link density in the docs.

```suggestion
:option:`!--host` argument - for example,
```

**Comment by freakboy3742** [comment]
The new buildbot configuration has been deployed; [this build](https://buildbot.python.org/all/#/builders/1382/builds/46) shows an auto detected iOS 13 minimum, and no build warnings (compared to the [previous PR build](https://buildbot.python.org/all/#/builders/1382/builds/45), which shows 2 warnings associated with the SQLite issue).

**Review by erlend-aasland — APPROVED** [review]
Looks good, with a tiny nit.

**Comment by freakboy3742** [comment]
!buildbot iOS

**Review by ned-deily — APPROVED** [review]

**Comment by freakboy3742** [comment]
!buildbot iOS

**Comment by zware** [comment]
!buildbot iOS

**Comment by zware** [comment]
!buildbot iOS

**Comment by zware** [comment]
Found how I broke it...

**Comment by freakboy3742** [comment]
!buildbot iOS

**Comment by zware** [comment]
!buildbot iOS

**Comment by freakboy3742** [comment]
!buildbot iOS

**Comment by freakboy3742** [comment]
For posterity - merged despite the CI failures. The CI failures are due to #121832; local testing and inspection of the CI logs demonstrates that the new minimum version is being applied.

**Pull request merged**